tests/config_files/config_testing.py

```diff
@@ -0,0 +1,615 @@
+        with open(Path(__file__).with_name('base_config.ini'), 'r') as f:
+            config = json.load(f)
+        config['n200s'] = [config['n200s'][0]]    # Only keep the first
+        n200 = config['n200s'][0]
+        n200['rx_channel_0'] = ''
+        n200['rx_channel_1'] = ''
+        n200['tx_channel_0'] = 'i0'   # cannot connect to intf antenna
+        with open(f'{os.environ["BOREALISPATH"]}/config/test/test_config.ini', 'w') as f:
+            json.dump(config, f)
+
+        with self.assertRaisesRegex(ValueError, "Cannot connect tx channel to interferometer array"):
+            MockOptions()
+
+    def testSingleN200TxRxDifferentAntennas(self):
+        """ Single N200, connected to different antennas for TX and RX (which is allowed) """
+        with open(Path(__file__).with_name('base_config.ini'), 'r') as f:
+            config = json.load(f)
+        config['n200s'] = [config['n200s'][0]]    # Only keep the first
+        n200 = config['n200s'][0]
+        n200['rx_channel_0'] = 'm0'
+        n200['rx_channel_1'] = ''
+        n200['tx_channel_0'] = 'm1'
+        with open(f'{os.environ["BOREALISPATH"]}/config/test/test_config.ini', 'w') as f:
+            json.dump(config, f)
+
+        options = MockOptions()
+        self.assertEqual(options.n200_count, 1)
+        self.assertEqual(options.n200_addrs, [n200['addr']])
+        self.assertEqual(len(options.rx_main_antennas), 1)
+        self.assertEqual(len(options.rx_intf_antennas), 0)
+        self.assertEqual(len(options.tx_main_antennas), 1)
+        self.assertEqual(options.rx_main_antennas, [0])
+        self.assertEqual(options.tx_main_antennas, [1])
+
+    def testTwoN200sOutOfOrder(self):
+        """ Two N200s, the first connected to a higher-index antenna than the second """
+        with open(Path(__file__).with_name('base_config.ini'), 'r') as f:
+            config = json.load(f)
+        config['n200s'] = config['n200s'][:2]    # Only keep the first two
+        n200_0 = config['n200s'][0]
+        n200_0['rx_channel_0'] = 'm1'
+        n200_0['rx_channel_1'] = ''
+        n200_0['tx_channel_0'] = 'm1'
+        n200_1 = config['n200s'][1]
+        n200_1['rx_channel_0'] = 'm0'
+        n200_1['rx_channel_1'] = ''
+        n200_1['tx_channel_0'] = 'm0'
+        with open(f'{os.environ["BOREALISPATH"]}/config/test/test_config.ini', 'w') as f:
+            json.dump(config, f)
+
+        options = MockOptions()
+        self.assertEqual(options.n200_count, 2)
+        self.assertEqual(options.n200_addrs, [n200_0['addr'], n200_1['addr']])
+        self.assertEqual(len(options.rx_main_antennas), 2)
+        self.assertEqual(len(options.rx_intf_antennas), 0)
+        self.assertEqual(len(options.tx_main_antennas), 2)
+        self.assertEqual(options.rx_main_antennas, [0, 1])
+        self.assertEqual(options.tx_main_antennas, [0, 1])
+
+    def testTooManyIntfAntennas(self):
+        """ Connected to all intf antennas, plus one invalid intf antenna """
```

Comment:
```suggestion
        """ Connected to all intf antennas, plus one invalid intf antenna. inft_antenna_count = 4 """
```
When I first saw these tests checking main/intf antenna indices higher than allowed I was slightly confused - I forgot that main/intf_antenna_count is specified in the config file.

Mentioning the config file settings (i.e. intf_antenna_count) that are being tested against in a given test would make these tests more clear, or make it clearer that all tests are tested against `base_config.ini`

Reply:
That's a good point, I will do that.